builtin/credential/aws/path_login.go

```diff
@@ -943,6 +948,10 @@ func (b *backend) pathLoginRenewIam(ctx context.Context, req *logical.Request, d
 		return nil, fmt.Errorf("role entry not found")
 	}
 
+	if !cidrutil.RemoteAddrIsOk(req.Connection.RemoteAddr, roleEntry.BoundCIDRs) {
```

Comment:
For renewal the token that is being used to renew will already have bound cidrs set on the token checked. I'm not sure we need to also check against the role. I guess it _probably_? Doesn't? Hurt?